frontend/vre/utils/jsonld.model.js

```diff
@@ -1,11 +1,102 @@
+ * it refers to if they are found in the graph.
+ * The subject passed to this function as an argument is not changed.
+ * @param subjectsByID{Dictionary<JSONLDSubject>} - The full contents of the graph in JSON-LD
+ * @param subject{JSONLDSubject} - The subject including its predicates and objects to create a nested version of
+ * @param parentSubjectIDs{Array<String>} - For internal use of recursive function; leave undefined
+ * @returns {Object}
+ */
+export function nestSubject(subjectsByID, subject, parentSubjectIDs=undefined) {
+    if (typeof parentSubjectIDs === "undefined") {
+        parentSubjectIDs = [subject["@id"]];
+    } else {
+        parentSubjectIDs = Array.from(parentSubjectIDs);
+        parentSubjectIDs.push(subject["@id"]);
+    }
+    const transformedSubject = _.clone(subject);
+    for (let property of Object.keys(subject)) {
+        if (subject[property].hasOwnProperty("@id")) {
+            // This is a reference to another subject
+            const refereedSubject = subjectsByID[subject[property]["@id"]];
+            if (refereedSubject && !(parentSubjectIDs.includes(refereedSubject["@id"]))) {
+                /* If the refereed subject was found in the graph, use it as replacement.
+                   Only do this if the subject is not the same as the one we started with,
+                   to avoid an endless loop. (Alternative would be to create a circular reference) */
+                transformedSubject[property] = nestSubject(subjectsByID, refereedSubject, parentSubjectIDs);
+            }
+        }
+    }
+    return transformedSubject;
+}
+
+/**
+ * Generic subclass of APICollection that parses incoming compacted JSON-LD to an
+ * ordered array of subjects according to the information of the
+ * `OrderedCollection` entity (ActivityStreams ontology) from the same graph.
+ * Sets the `totalResults` attribute if available.
+ * The graph should contain exactly one `OrderedCollection`.
+ * @class
+ */
+export var JsonLdWithOCCollection = APICollection.extend({
+    model: JsonLdModel,
+    /**
+     * The total number of results. This is filled by `parse` if the
+     * `OrderedCollection` subject comes with `totalItems`.
+     * @type {?number}
+     */
+    totalResults: undefined,
+    /**
+     * The prefix, used in JSON-LD, for the ActivityStreams namespace.
+     * Defaults to `as:` but can be overridden.
+     * @type {string}
+     */
+    activityStreamsPrefix: "as:",
+    parse: function(response) {
+        // Get all subjects of the graph with their predicates and objects as an array
+        if (!response.hasOwnProperty("@graph")) {
+            throw "Response has no @graph key, is this JSON-LD in compacted form?";
+        }
+        const allSubjects = response["@graph"];
+        const orderedCollection = allSubjects.find((subject) => {return subject["@type"] === `${this.activityStreamsPrefix}OrderedCollection`}, this);
```

Comment:
I think I forgot to mention this in my previous review, but Underscore's `_.find` lets you write  a nice shorthand:

```suggestion
        const orderedCollection = _.find(allSubjects, {"@type": `${this.activityStreamsPrefix}OrderedCollection`});
```

With an additional variable, the code is even easier to read:

```suggestion
        const ocType = `${this.activityStreamsPrefix}OrderedCollection`;
        const orderedCollection = _.find(allSubjects, {"@type": ocType});
```